test/e2e/network/loadbalancer.go

```diff
@@ -1236,6 +1244,143 @@ var _ = common.SIGDescribe("LoadBalancers", func() {
+		err = wait.PollImmediate(dsRetryPeriod, dsRetryTimeout, e2edaemonset.CheckDaemonPodOnNodes(f, ds, nodeNames))
+		framework.ExpectNoError(err, "error waiting for daemon pods to start")
+		err = e2edaemonset.CheckDaemonStatus(f, name)
+		framework.ExpectNoError(err)
+
+		ginkgo.By(fmt.Sprintf("Creating a service %s with type=LoadBalancer in namespace %s", name, ns))
+		jig := e2eservice.NewTestJig(cs, ns, name)
+		jig.Labels = labels
+		service, err := jig.CreateLoadBalancerService(e2eservice.GetServiceLoadBalancerCreationTimeout(cs), func(svc *v1.Service) {})
+		framework.ExpectNoError(err)
+
+		lbNameOrAddress := e2eservice.GetIngressPoint(&service.Status.LoadBalancer.Ingress[0])
+		svcPort := int(service.Spec.Ports[0].Port)
+
+		ginkgo.By("Hitting the DaemonSet's pods through the service's load balancer")
+		timeout := e2eservice.LoadBalancerLagTimeoutDefault
+		if framework.ProviderIs("aws") {
+			timeout = e2eservice.LoadBalancerLagTimeoutAWS
+		}
+		e2eservice.TestReachableHTTP(lbNameOrAddress, svcPort, timeout)
+
+		ginkgo.By("Starting a goroutine to continuously hit the DaemonSet's pods through the service's load balancer")
+		var totalRequests uint64 = 0
+		var networkErrors uint64 = 0
+		var httpErrors uint64 = 0
+		done := make(chan struct{})
+		defer close(done)
+		go func() {
+			defer ginkgo.GinkgoRecover()
+
+			wait.Until(func() {
+				atomic.AddUint64(&totalRequests, 1)
+				client := &http.Client{
+					Transport: utilnet.SetTransportDefaults(&http.Transport{
+						DisableKeepAlives: true,
+					}),
+					Timeout: 5 * time.Second,
+				}
+				ipPort := net.JoinHostPort(lbNameOrAddress, strconv.Itoa(svcPort))
+				msg := "hello"
+				url := fmt.Sprintf("http://%s/echo?msg=%s", ipPort, msg)
+				resp, err := client.Get(url)
+				if err != nil {
+					framework.Logf("Got error testing for reachability of %s: %v", url, err)
+					atomic.AddUint64(&networkErrors, 1)
+					return
+				}
+				defer resp.Body.Close()
+				if resp.StatusCode != http.StatusOK {
+					framework.Logf("Got bad status code: %d", resp.StatusCode)
+					atomic.AddUint64(&httpErrors, 1)
+					return
+				}
+				body, err := io.ReadAll(resp.Body)
+				if err != nil {
+					framework.Logf("Got error reading HTTP body: %v", err)
+					atomic.AddUint64(&httpErrors, 1)
+					return
+				}
+				if string(body) != msg {
+					framework.Logf("The response body does not contain expected string %s", string(body))
+					atomic.AddUint64(&httpErrors, 1)
+					return
+				}
+			}, time.Duration(0), done)
+		}()
+
+		ginkgo.By("Triggering DaemonSet rolling update several times")
+		var minSuccessRate float64 = 0.95
+		var previousTotalRequests uint64 = 0
+		var previousNetworkErrors uint64 = 0
+		var previousHttpErrors uint64 = 0
+		retryTimeout := dsRetryTimeout + time.Duration(len(nodeNames)*30)*time.Second // time to complete the rolling update is proportional to the number of nodes in the cluster
```

Comment:
this is going to be a problem on large scale testing 5k, 15k @wojtek-t @mborsz can you advice here on how to make this test in a way it doesn't break at scale?